components/kyma-environment-broker/internal/process/steps/lifecycle_manager_kubeconfig.go

```diff
@@ -0,0 +1,96 @@
+package steps
+
+import (
+	"context"
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// syncKubeconfig step ensures desired state of kubeconfig secret for lifecycle manager
+type syncKubeconfig struct {
+	k8sClient client.Client
+	cleanup   bool
+}
+
+func SyncKubeconfig(k8sClient client.Client) syncKubeconfig {
+	return syncKubeconfig{
+		k8sClient: k8sClient,
+	}
+}
+
+func DeleteKubeconfig(k8sClient client.Client) syncKubeconfig {
+	return syncKubeconfig{
+		k8sClient: k8sClient,
+		cleanup:   true,
+	}
+}
+
+func (_ syncKubeconfig) Name() string {
+	return "Sync_Kubeconfig"
+}
+
+func (s syncKubeconfig) Run(o internal.Operation, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	if s.cleanup {
```

Comment:
the only one thing which is common between steps for ensuring and cleaning up is the k8sClient. everything else is different. I do not see any reason to have it in one "class" and a flag which decides which logic must be run. I'd prefer two implementations (two structs). It could be even in a one file.